Mathlib/Data/Fintype/Pi.lean

```diff
@@ -157,6 +157,10 @@ noncomputable instance _root_.Function.Embedding.fintype {α β} [Fintype α] [F
   classical exact Fintype.ofEquiv _ (Equiv.subtypeInjectiveEquivEmbedding α β)
 #align function.embedding.fintype Function.Embedding.fintype
 
+noncomputable instance _root_.RelEmbedding.instFintype {α β} [Fintype α] [Fintype β]
+    {r : α → α → Prop} {s : β → β → Prop} : Fintype (r ↪r s) :=
+  Fintype.ofInjective _ RelEmbedding.toEmbedding_injective
```

Comment:
This is a bad instance since it causes a non-defeq diamond with the existing `SimpleGraph.Hom.instFintype`. Can you generalise `SimpleGraph.Hom.instFintype` to arbitrary `RelHom` instead?